File: codesignal/arcade/2.core/9.well of integration/2.allLongestStrings.cpp
// Given an array of strings, return another array containing all of its longest strings.

#include <vector>
#include <string>

class Solution {
public:
  std::vector<std::string> allLongestStrings(std::vector<std::string> a) {
    std::vector<std::string> r;
    int max = 0;
    for (auto i : a) {
      if (i.size() > max) {
        max = i.size();
        r.clear();
        r.push_back(i);
      } else if (i.size() == max) {
        r.push_back(i);
      }
    }
    return r;
  }
};

File: codesignal/arcade/2.core/9.well of integration/2.allLongestStrings.py
# Given an array of strings, return another array containing all of its longest strings.

class Solution:
    def allLongestStrings(self, inputArray):
        return [x for x in inputArray if len(x) == max([len(y) for y in inputArray])]
    
class Solution:
    def allLongestStrings(self, inputArray):
        r = []
        l = max(inputArray, key=len)
        for s in inputArray:
            if len(s) == len(l):
                r.append(s)
        return r